src/ui/common/form/RenderDateTimePickerInput.js

```diff
@@ -0,0 +1,144 @@
+  }
+
+  render() {
+    const {
+      input, name, label, help, locale, dateFormat, placeholder,
+      meta: { touched, error }, hoursList, minutesList, secondsList,
+      isClearable, hasLabel,
+    } = this.props;
+    const optionsHours = hoursList.map(item => padStart(item, 2, '0'));
+    const optionsMinutes = minutesList.map(item => padStart(item, 2, '0'));
+    const optionsSeconds = secondsList.map(item => padStart(item, 2, '0'));
+
+    return (
+      <div className="form-group">
+        {hasLabel && (
+          <label htmlFor={name} className="col-xs-2 control-label">
+            {label} {help}
+          </label>
+        )}
+        <Col xs={10}>
+          <Row>
+            <Col xs={3}>
+              <DatePicker
+                {...input}
+                value={input.value.date}
+                placeholder={placeholder}
+                selected={input.value.date ? moment(input.value.date, dateFormat) : null}
+                onChange={this.handleChange}
+                disabledKeyboardNavigation
+                locale={locale}
+                dateFormat={dateFormat}
+                isClearable={isClearable}
+                calendarClassName="RenderDatePickerInput__calendar"
+              />
+            </Col>
+            <Col xs={3}>
+              <Field
+                name={`${input.name}.hours`}
+                component="select"
+                className="form-control"
+              >
+                { optionsHours.map(value => (<option key={value}>{value}</option>)) }
+              </Field>
+              <span className="help help-block">
+                <FormattedMessage id="cms.datetimepicker.label.hours" />
+              </span>
+            </Col>
+            <Col xs={3}>
+              <Field
+                name={`${input.name}.minutes`}
+                component="select"
+                className="form-control"
+              >
+                { optionsMinutes.map(value => (<option key={value}>{value}</option>)) }
+              </Field>
+              <span className="help help-block">
+                <FormattedMessage id="cms.datetimepicker.label.minutes" />
+              </span>
+            </Col>
+            <Col xs={3}>
+              <Field
+                name={`${input.name}.seconds`}
+                component="select"
+                className="form-control"
+              >
+                { optionsSeconds.map(value => (<option key={value}>{value}</option>)) }
+              </Field>
+              <span className="help help-block">
+                <FormattedMessage id="cms.datetimepicker.label.seconds" />
+              </span>
+            </Col>
+          </Row>
+          <Col xs={12}>
+            <div className="help-block help-block-error">
+              {touched ? error : ''}
+            </div>
+          </Col>
+        </Col>
+      </div>
+    );
+  }
+}
+
+RenderDateTimePickerInput.propTypes = {
+  onWillMount: PropTypes.func,
```

Comment:
Please remove it if its unused. Also from defaultProps. Also a general rule is to use onDidMount rather than onWillMount

Reply:
I agree. I actually just borrowed the code from app-builder.